src/Hamiltonians/FroelichPolaron_nD.jl

```diff
@@ -0,0 +1,205 @@
+using SpecialFunctions
+
+#structure 
+struct FroehlichPolaron_nD{
+    T,
+    M,
+    D,
+    A<:OccupationNumberFS{M}, 
+    MC,
+    G<:CubicGrid
+    } <: AbstractHamiltonian{T}
+    address::A
+    d:: Int64
+    geometry::G
+    alpha::T
+    mass::T
+    omega::T
+    l::SVector{D, Float64}
+    p::SVector{D, Float64}
+    ks::SVector{M, SVector{D, T}}
+    momentum_cutoff::Union{MC, Nothing}
+    mode_cutoff::Union{Int, Nothing}
+end
+
+#function for nD Froehlich Polaron
+function FroehlichPolaron_nD(
+    address::OccupationNumberFS{M,AT};
+    D::Int = 2,
+    geometry::CubicGrid = PeriodicBoundaries(ntuple(_ -> Int(round(num_modes(address)^(1/D))), D)),
+    alpha = 1,
+    mass = 1,
+    omega = 1,
+    l = ones(D),
+    p = zeros(D),
+    momentum_cutoff::Union{Nothing,Real} = nothing,
+    mode_cutoff::Union{Nothing,Int} = nothing,
+) where {M, AT}
+
+    # Check positive l
+    if length(l) != D || any(x -> x <= 0, l)
+        throw(ArgumentError("`l` must be a positive vector of length $D"))
+    end
+
+    #check m_lin is possible
+    if abs(M^(1/D) - round(M^(1/D))) >0.01  
```

Comment:
```suggestion
    if abs(M^(1/D) - round(M^(1/D))) > 0.01  
```
Try to put spaces on both sides of operators, except in very long expressions where having some terms close together makes it more readable.